core/trino-main/src/main/java/io/trino/server/ServerConfig.java

```diff
@@ -104,4 +121,10 @@ public ServerConfig setQueryInfoUrlTemplate(String queryInfoUrlTemplate)
         this.queryInfoUrlTemplate = Optional.ofNullable(queryInfoUrlTemplate);
         return this;
     }
+
+    public enum CoordinatorShutdownStrategy
+    {
+        CANCEL_RUNNING, // Cancel running queries after grace period has passed
```

Comment:
The name is misleading, I initially thought it cancels all running queries immediately. Should there be a third option to do this? It would inform clients the server is either being restarted so they can retry, or shut down so they don't have to keep trying.